client/client-multi/client-c/src/kaa/platform-impl/common/encryption_utils.c

```diff
@@ -0,0 +1,186 @@
+/*
+ *  Copyright 2014-2016 CyberVision, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+#include "platform/ext_encryption_utils.h"
+#include "platform/ext_key_utils.h"
+
+#include <assert.h>
+#include <stdbool.h>
+#include <string.h>
+
+#include <utilities/kaa_aes_rsa.h>
+
+#include <mbedtls/pk.h>
+#include <mbedtls/entropy.h>
+#include <mbedtls/ctr_drbg.h>
+#include <mbedtls/md.h>
+
+#define ENC_SESSION_KEY_LENGTH         256
+#define KAA_SIGNATURE_LENGTH           256
+#define AES_ECB_ENCRYPTION_CHUNCK_SIZE 16
+
+extern mbedtls_pk_context pk_context_;
+
+/**
+ * Contains Endpoint keys.
+ *
+ * The main purpose of the structure is
+ * caching calculated keys.
+ */
+static struct {
+    uint8_t session_key[KAA_SESSION_KEY_LENGTH];
+    uint8_t encrypted_session_key[ENC_SESSION_KEY_LENGTH];
+    uint8_t signature[KAA_SIGNATURE_LENGTH];
+    size_t  encrypted_session_key_length;
```

Comment:
Hell yeah, the size of RSA-encrypted data is always equal to the RSA key length (2048 bits in our case)